std/assembly/typedarray.ts

```diff
@@ -39,6 +40,10 @@ export class Int8Array extends TypedArray<i8> {
   ): T {
     return REDUCE_RIGHT<Int8Array, i8, T>(this, callbackfn, initialValue);
   }
+
+  map(callbackfn: (value: i8, index: i32, self: this) => i8): this {
+    return MAP<this, i8>(this, callbackfn);
```

Comment:
Let's try to avoid `this` for now because this might lead to multiple compilation of the same function, just differing `this` type, for subclasses, depending on how we decide to go forward with that.

Reply:
Actually using `this` on map is incorrect. `this` as type should using only for mutable functions which explicitly return `this` reference

Reply:
Might it make sense to limit `: this` return type'd functions to actual `return this` statements and otherwise emit an error?

Reply:
Okay. I don't mind making the change. I'll submit in just a moment.

Reply:
Hmm, may be better use custom lint for that?

Reply:
@MaxGraey I see. I used vsstudio to write the function using the multicaret editing and `this` was just an easy way to do it. I went through and implemented the class type for each method. It should be fixed now. Thanks for your help.